lib/src/main/java/ECSE428Project/service/LogoutService.java

```diff
@@ -19,29 +19,42 @@
     private AccountRepository accountRepository;
 
 
-    //Do we need to implement a "find current logged in account" method?
+    //It would be good to implement a "find current logged in account" method
     //Could be implemented in the library class
+/*
+    //Retrieve all account entities saved in the accountRepository
+    List<Account> savedAccounts = accountRepository.findAll();
+
+    if(savedAccounts.isEmpty()) {
+        throw new Exception...
+    }
+    //Retrieve the first entity in the account list
+    Account dummyAccount = savedAccounts.get(0);
+*/
 
 
     @Transactional
     public Account profileLogout(String email) {
 
-        Account profile = accountRepository.findAccountByEmail(email);
+        if (email == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Email cannot be empty! Please Enter Valid Credentials");
+        }
 
+        Account profile = accountRepository.findAccountByEmail(email);
```

Comment:
You should probably use an Optional here to make sure that we don't run into any null pointer exceptions.
You could use this code:

        if (email == null) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Email cannot be empty! Please Enter Valid Credentials");
        }        
        Optional<Account> optAccount = accountRepository.findById(email);
        Account profile;

        // Check that the account exists in the database
        if (optAccount.isPresent()) {
            profile = optAccount.get();
        } else {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,"No account associated to the provided credentials"); 
       }

       if (profile.isLoggedIn(), etc....